core/src/pow/types.rs

```diff
@@ -391,21 +391,39 @@ impl Proof {
 	}
 }
 
-#[inline(always)]
+fn extract_bits(bits: &Vec<u8>, bit_start: usize, bit_count: usize, read_from: usize) -> u64 {
+	let mut buf: [u8; 8] = [0; 8];
```

Comment:
Is it possible in rust to directly assign

let bits64 = u64::from_le_bytes(&bits[read_from..read_from + 8]) ?

if so, that would be much preferable over the extra buffer and copying....

Reply:
Other than that, this PR looks fine to me.

Reply:
Array and slice are different types in Rust, so we can't directly use `bits` in `from_le_bytes`

Reply:
and there's no function or method we can call on the slice to convert to array?

Reply:
btw, remaining comments are just about aesthetics (can always be reconsidered later) and should not hold up this PR...